packages/client/hmi-client/src/api/api.ts

```diff
@@ -181,4 +186,82 @@ export class Poller<T> {
 	}
 }
 
+/**
+ * Interface defining handlers for Server-Sent Events (SSE).
+ */
+export interface SSEHandlers {
+	/**
+	 * Handler for incoming SSE messages.
+	 * @param {any} message The received message.
+	 * @param {() => void} abort Function to abort the SSE connection.
+	 */
+	onMessage: (message: any, abort: () => void) => void;
+	/**
+	 * Handler for SSE connection open event.
+	 * @param {Response} response The response object.
+	 */
+	onOpen?: (response: Response) => void;
+	/**
+	 * Handler for SSE connection error.
+	 * @param {Error} error The error object.
+	 */
+	onError: (error: Error) => void;
+	/**
+	 * Handler for SSE connection close event.
+	 */
+	onClose?: () => void;
+}
+
+/**
+ * Function to handle Server-Sent Events (SSE) connections.
+ * @param {string} url The URL to connect to.
+ * @param {SSEHandlers} handlers Object containing SSE event handlers.
+ * @param {FetchEventSourceInit} options Additional options for the SSE connection.
+ */
+export async function handleSSE(
+	url: string,
+	handlers: SSEHandlers,
+	options?: FetchEventSourceInit
+) {
+	const authStore = useAuthStore();
+	const controller = new AbortController();
+	const fetchEventSourceOptions: FetchEventSourceInit = {
+		...options,
+		headers: {
+			...options?.headers,
+			Authorization: `Bearer ${authStore.token}`
+		},
+		onmessage(message: EventSourceMessage) {
+			const data = message?.data;
+			const abort = () => controller.abort();
+			handlers.onMessage(data, abort);
```

Comment:
Why use CamelCase for the handlers method, if the goal is to replicate the SSE ones?

Reply:
there wasn't really intention behind it Ive changed it to flatcase